packages/vite/src/node/config.ts

```diff
@@ -1144,10 +1144,16 @@ async function loadConfigFromBundledFile(
   // convert to base64, load it with native Node ESM.
   if (isESM) {
     try {
+      // Postfix the bundled code with a timestamp to avoid Node's ESM loader cache
+      const configTimestamp = `${fileName}.timestamp:${Date.now()}-${Math.random()
+        .toString(16)
+        .slice(2)}`
       return (
         await dynamicImport(
           'data:text/javascript;base64,' +
-            Buffer.from(bundledCode).toString('base64'),
+            Buffer.from(`${bundledCode}\n//${configTimestamp}`).toString(
+              'base64',
+            ),
```

Comment:
I wonder if this would affect sourcemaps since we're bundling with esbuild with `sourcemap: 'inline'`, but I'm also not sure if we're using the sourcemap in anyway 🤔 

Other than that this looks good to me though, nice find with the issue! I also wonder if doing a `{ ...config }` would also fix the issue if it's an object.